src/pages/settings/Profile/ProfilePage.js

```diff
@@ -190,17 +184,46 @@ class ProfilePage extends Component {
         }, true);
     }
 
-    validateInputs() {
-        const [hasFirstNameError, hasLastNameError, hasPronounError] = ValidationUtils.doesFailCharacterLimit(
-            50,
-            [this.state.firstName.trim(), this.state.lastName.trim(), this.state.pronouns.trim()],
-        );
-        this.setState({
-            hasFirstNameError,
-            hasLastNameError,
-            hasPronounError,
-        });
-        return !hasFirstNameError && !hasLastNameError && !hasPronounError;
+    validate(values) {
+        const errors = {};
+
+        if (values.firstName) {
+            const [hasFirstNameError] = ValidationUtils.doesFailCharacterLimit(
+                50,
+                [values.firstName.trim()],
+            );
+
+            if(hasFirstNameError) {
+                this.setState({hasFirstNameError});
+                errors.firstName = PersonalDetails.getMaxCharacterError(this.state.hasFirstNameError);
+            }
+        }
+
+        if (values.lastName) {
```

Comment:
It is necessary to check because `values.firstName` would `undefined` if the Firstname is not modified and the statement `values.firstName.trim()` would result into a JavaScript runtime error

<img width="567" alt="image" src="https://user-images.githubusercontent.com/911879/174352053-0cd0725c-e04e-41c0-a666-b678118231a8.png">

Reply:
Ahh makes sense!

What do you say we create a new function `ValidationUtils.doesFailCharacterLimitAfterTrim()`?
It would make the code look more pretty, and I'm sure other pages will also use this function when they're refactored to use FORM.

Reply:
I think that's a good refactoring, will make the `ProfilePage` comparatively cleaner and like you shared, can be useful in other pages as well.

Would you like me to go ahead and add a new function to `ValidationUtils`?

Reply:
Yeah sure!

Reply:
I updated the `doesFailCharacterLimitAfterTrim()`  from 
```
function doesFailCharacterLimit(maxLength, valuesToBeValidated) {
    return _.map(valuesToBeValidated, value => value.length > maxLength);
}
```
to
```
function doesFailCharacterLimit(maxLength, valuesToBeValidated) {
    return _.map(valuesToBeValidated, value => value && value.trim().length > maxLength);
}
```

After the modification, the use of `trim()` can be removed from the `ProfilePage` plus the `if` check you shared in one of your comment above can also be removed. Following is how the `ProfilePage` changes looks:

```
const [hasFirstNameError] = ValidationUtils.doesFailCharacterLimit(
            CONST.PROFILE_INPUTS_CHARACTER_LIMIT,
            [values.firstName],
        );

        if (hasFirstNameError) {
            this.setState({hasFirstNameError});
            errors.firstName = PersonalDetails.getMaxCharacterError(this.state.hasFirstNameError);
        }

        const [hasLastNameError] = ValidationUtils.doesFailCharacterLimit(
            CONST.PROFILE_INPUTS_CHARACTER_LIMIT,
            [values.lastName],
        );

        if (hasLastNameError) {
            this.setState({hasLastNameError});
            errors.lastName = PersonalDetails.getMaxCharacterError(this.state.hasLastNameError);
        }

        const [hasPronounError] = ValidationUtils.doesFailCharacterLimit(
            CONST.PROFILE_INPUTS_CHARACTER_LIMIT,
            [values.pronouns],
        );

        if (hasPronounError) {
            this.setState({hasPronounError});
            errors.pronouns = PersonalDetails.getMaxCharacterError(this.state.hasPronounError);
        }
```

Does this look okay? If you don't have any suggestions or modifications to this, I can push the changes.

Reply:
Actually, I was further able to refactor the `ProfilePage` code. Following is what I've now got:

```
const [hasFirstNameError, hasLastNameError, hasPronounError] = ValidationUtils.doesFailCharacterLimit(
    CONST.PROFILE_INPUTS_CHARACTER_LIMIT,
    [this.state.firstName, this.state.lastName, this.state.pronouns],
);

this.setState({
    hasFirstNameError,
    hasLastNameError,
    hasPronounError,
});

if (hasFirstNameError) {
    errors.firstName = PersonalDetails.getMaxCharacterError(this.state.hasFirstNameError);
}

if (hasLastNameError) {
    errors.lastName = PersonalDetails.getMaxCharacterError(this.state.hasLastNameError);
}

if (hasPronounError) {
    errors.pronouns = PersonalDetails.getMaxCharacterError(this.state.hasPronounError);
}
```

Reply:
Fantastic! 🙌

Reply:
Pushed the changes!